src/test/lib/storage/index/group_key/variable_length_key_test.cpp

```diff
@@ -49,4 +49,62 @@ TEST_F(VariableLengthKeyTest, CreateKeysWithOrAndShift) {
   EXPECT_TRUE(key == _key_reference);
 }
 
+
+/**
+ * Test that copy‐assignment correctly duplicates the bit pattern
+ * (this also covers operator=(const VariableLengthKeyBase&)).
+ */
+TEST_F(VariableLengthKeyTest, CopyAssignment) {
+  const auto bytes = CompositeKeyLength{8};
+
+  auto original = VariableLengthKey{bytes};
+  original <<= CompositeKeyLength{4};
+  original |= uint64_t{0xFFFFFFFFu};
```

Comment:
Could you please explain why this is necessary (an maybe find a way to ensure that `original` differs from the initialized `copy` below that I can still understand)?